server/examples.js

```diff
@@ -153,57 +153,112 @@ function getSketchContent(projectsInAllCategories) {
+                _id: b,
+                fileType: 'file',
+                children: []
+              },
+              {
+                name: 'style.css',
+                content: defaultCSS,
+                id: c,
+                _id: c,
+                fileType: 'file',
+                children: []
+              }
+            ],
+            _id: shortid.generate()
+          });
 
-    projectsInAllCategories.forEach(projectsInOneCategory => {
-      projectsInOneCategory.forEach(project => {
-        const newProject = new Project({
-          name: project.projectName,
-          user: user._id,
-          files: [
-            {
-              name: 'root',
-              id: r,
-              _id: r,
-              children: [a, b, c],
-              fileType: 'folder'
-            },
-            {
-              name: 'sketch.js',
-              content: project.sketchContent,
-              id: a,
-              _id: a,
-              isSelectedFile: true,
-              fileType: 'file',
-              children: []
-            },
-            {
-              name: 'index.html',
-              content: defaultHTML,
-              id: b,
-              _id: b,
-              fileType: 'file',
-              children: []
-            },
-            {
-              name: 'style.css',
-              content: defaultCSS,
-              id: c,
-              _id: c,
-              fileType: 'file',
-              children: []
+          let assetsInProject = project.sketchContent.match(/assets\/[\w-]+\.[\w]*/g) || project.sketchContent.match(/assets\/[\w-]*/g) || [];
+          // if (assetsInProject === []) {
+          //   assetsInProject = project.sketchContent.match(/assets\/[\w-]*/g) || [];
+          // }
+
+          assetsInProject.forEach((assetName, i) => {
+            assetName = assetName.split('assets/')[1];
+
+            assets.forEach(asset => {
+              if (asset.name === assetName || asset.name.split('.')[0] === assetName) {
+                assetName = asset.name;
+              }
+            });
+            if (i === 0) {
+              const id = objectID().toHexString();
+              newProject.files.push({
+                name: 'assets',
+                id,
+                _id: id,
+                children: [],
+                fileType: 'folder'
+              });
+              // add assets folder inside root
+              newProject.files[0].children.push(id);
             }
-          ],
-          _id: shortid.generate()
-        });
 
-        newProject.save( (err, newProject) => {
-          if (err) throw err;
-          console.log('Created a new project in p5 user: ' + newProject.name);
-        });
+            const fileID = objectID().toHexString();
+            newProject.files.push({
+              name: assetName,
+              url: `https://raw.githubusercontent.com/processing/p5.js-website/master/dist/assets/examples/assets/${assetName}?client_id=${
```

Comment:
i just tried it, and you don't need to include your github credentials in the url, e.g. https://raw.githubusercontent.com/processing/p5.js-website/master/dist/assets/examples/assets/mask.png works!